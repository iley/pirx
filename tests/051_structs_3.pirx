struct Vector {
  x: int;
  y: int;
}

func vector_add(v1: *Vector, v2: *Vector, out: *Vector) {
  (*out).x = (*v1).x + (*v2).x;
  (*out).y = (*v1).y + (*v2).y;
}

func vector_print(v: *Vector) {
  printf("(%d, %d)\n", (*v).x, (*v).y);
}

func main(): int {
  var v1: Vector;
  v1.x = 1;
  v1.y = 2;

  var v2: Vector;
  v2.x = 10;
  v2.y = 20;

  var res: Vector;
  vector_add(&v1, &v2, &res);

  vector_print(&v1);
  vector_print(&v2);
  vector_print(&res);

  return 0;
}
