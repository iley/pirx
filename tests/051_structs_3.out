(1, 2)
(10, 20)
(11, 22)
